Compilation Failure on `GOOS=aix`

We have received a report of compilation failure with `GOOS` is `aix` (see caddyserver/caddy#5970). The build failure reports the following:

```
# github.com/google/certificate-transparency-go/x509
../../../../go/pkg/mod/github.com/google/certificate-transparency-go@v1.1.6/x509/cert_pool.go:65:9: undefined: loadSystemRoots
../../../../go/pkg/mod/github.com/google/certificate-transparency-go@v1.1.6/x509/root.go:21:32: undefined: loadSystemRoots
../../../../go/pkg/mod/github.com/google/certificate-transparency-go@v1.1.6/x509/verify.go:759:12: c.systemVerify undefined (type *Certificate has no field or method systemVerify)
```

The 3 errors listed in the error message report functions/methods that are defined in `root_unix.go`. The [build tags](https://github.com/google/certificate-transparency-go/blob/df496fc651394b4aa5ab529b8bc35010c3d282e4/x509/root_unix.go#L5-L6) exclude `AIX`. I believe the extensive list can be replaced with the new `unix` build tag (implemented here https://github.com/golang/go/commit/c3fcd0117784ac05e35120ac0de6c960b400a31e) that will include any Unix-like system by default. This avoids such hassles in the future. FWIW, AIX supports [`SSL_CERT_DIR`](https://www.ibm.com/docs/en/search/SSL_CERT_DIR?scope=ssw_aix_73) and [`SSL_CERT_FILE`](https://www.ibm.com/docs/en/search/SSL_CERT_FILE?scope=ssw_aix_73) if you prefer to be explicit about the supported OS.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -2,6 +2,10 @@
 
 ## HEAD
 
+### Add support for AIX
+
+* Add build tags for AIX operating system
+
 ## v1.1.7
 
 * Recommended Go version for development: 1.20

--- a/x509/root_aix.go
+++ b/x509/root_aix.go
@@ -1,0 +1,10 @@
+// Copyright 2018 The Go Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package x509
+
+// Possible certificate files; stop after finding one.
+var certFiles = []string{
+	"/var/ssl/certs/ca-bundle.crt",
+}

--- a/x509/root_unix.go
+++ b/x509/root_unix.go
@@ -2,8 +2,8 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
-// +build dragonfly freebsd linux netbsd openbsd solaris zos
+//go:build aix || dragonfly || freebsd || linux || netbsd || openbsd || solaris || zos
+// +build aix dragonfly freebsd linux netbsd openbsd solaris zos
 
 package x509
 
@@ -19,6 +19,7 @@ var certDirectories = []string{
 	"/usr/local/share/certs",       // FreeBSD
 	"/etc/pki/tls/certs",           // Fedora/RHEL
 	"/etc/openssl/certs",           // NetBSD
+	"/var/ssl/certs",               // AIX
 }
 
 const (

